src/simd/distances_sse.cc

```diff
@@ -67,20 +68,12 @@ fvec_norm_L2sqr_sse(const float* x, size_t d) {
 
 float
 fp16_vec_norm_L2sqr_sse(const knowhere::fp16* x, size_t d) {
-    __m128 m_res = _mm_setzero_ps();
-    while (d >= 4) {
-        __m128 m_x = _mm_cvtph_ps(_mm_loadl_epi64((const __m128i_u*)x));
-        m_res = _mm_add_ps(m_res, _mm_mul_ps(m_x, m_x));
-        x += 4;
-        d -= 4;
+    std::vector<knowhere::fp32> fp32_x(d);
```

Comment:
`std::vector` introduces an enormous overhead.
Please implement these functions manually in the following way:
```
    float res = 0;
    for (size_t i = 0; i < d; i++) {
        const float xv = float(x[i]);
        res += xv * xv;
    }

    return res;
``` 
Similar for dot products